google-cloud-storage/src/main/java/com/google/cloud/storage/Storage.java

```diff
@@ -3547,4 +3547,43 @@ HmacKeyMetadata updateHmacKeyState(
    * @throws StorageException upon failure
    */
   ServiceAccount getServiceAccount(String projectId);
+
+  /**
+   * Creates a notification subscription for a given bucket.
+   *
+   * @param bucket name of the bucket
+   * @param notification a {@code Notification} object
+   * @return the notification that was created
+   * @throws StorageException upon failure
+   */
+  Notification createNotification(String bucket, Notification notification);
+
+  /**
+   * Gets the notification with the specified name on the bucket.
+   *
+   * @param bucket name of the bucket
+   * @param notification notification ID
+   * @return the notification object that exist on the bucket or {@code null} if not found
+   * @throws StorageException upon failure
+   */
+  Notification getNotification(String bucket, String notification);
+
+  /**
+   * Retrieves a list of notification subscriptions for a given bucket.
+   *
+   * @param bucket name of the bucket
+   * @return a list of {@link Notification} objects that exist on the bucket.
+   * @throws StorageException upon failure
+   */
+  List<Notification> listNotifications(String bucket);
+
+  /**
+   * Deletes the notification subscription with the specified name on the bucket.
+   *
+   * @param bucket name of the bucket
+   * @param notification ID of the notification to delete
+   * @return {@code true} if the notification has been deleted, {@code false} if not found
+   * @throws StorageException upon failure
+   */
+  boolean deleteNotification(String bucket, String notification);
```

Comment:
please rename `notification` param to make it clear what it stands for. id, name, or topic.